### ISSUE TITLE
Question about preparing the training data. 

### ISSUE THREAD

**Issue by XuyangBai** [mihaidusmanu/d2-net#14]
Hi, Thanks for your sharing. I have a question about your paper.  In 4.3 part Implementation details, you mentioned that 

> For each pair, we selected a random 256 × 256 crop centered around one correspondence. We use a batch size of 1 and make sure that the training pairs present at least 128 correspondences in order to obtain meaningful gradients.

In my understanding, your model needs pairs of images and pixel-wise correspondences as input. And then you densely extract the feature vector for each pixel and calculate the loss based on the correspondence information. Then why you need to crop the images insteaded of use the origin image? Is is because the memory issue ?  and what do you mean by `a random 256 * 256 crop centered around one correspondence?` around which correspondence? 

After that, do you choose a fixed number of pixel-wise correspondence as your positive pair or use all of them?  and how can you guarantee this two cropped images have at least 128 correspondence pixel pairs ?

Thanks a lot !

**Comment by mihaidusmanu** [comment]
Hello. Please find the answers inlined below.

> Then why you need to crop the images insteaded of use the origin image? 

You are right, we are using crops due to memory reasons mainly. Another reason for using crops is that training images have a wide range of resolutions. 

>  and what do you mean by a random 256 * 256 crop centered around one correspondence? around which correspondence?

The pipeline for dataset selection first picks a correspondence from the sparse 3D model and then crops a 256x256 area around it.

> After that, do you choose a fixed number of pixel-wise correspondence as your positive pair or use all of them? and how can you guarantee this two cropped images have at least 128 correspondence pixel pairs ?

We use **all pixel-wise correspondences** for computing the loss. Since we are using a batch size of 1, we simply skip the image pairs with fewer than 128 correspondences.

https://github.com/mihaidusmanu/d2-net/blob/be09ec72566968dfce96de577c5880a7aafdeace/lib/loss.py#L75-L76

**Comment by XuyangBai** [comment]
Thanks a lot !
